From 4e1dab773dcd6808b139beb59f551a1aa711cac4 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 22 May 2024 10:02:05 +0000
Subject: [PATCH] Revert "[SW-15774] package: extract agent-hbm from zephyr"

This breaks all the margin tool agents running on the device.
Revert for now, once fixed will be reapplied.

This reverts commit acd42b51c2c9a8287b95a2636881d215839237e1.

Change-Id: I1001dc1cac35fd3e2c9a94dc4e1428351458138d
---

diff --git a/Config.in b/Config.in
index 794cefa..8cf90a8 100644
--- a/Config.in
+++ b/Config.in
@@ -11,7 +11,6 @@
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/pid/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/mgmt/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/event/Config.in"
-source "$BR2_EXTERNAL_HABANALABS_PATH/package/agent-hbm/Config.in"
 
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/hl-fw-loader/Config.in"
 source "$BR2_EXTERNAL_HABANALABS_PATH/package/hls2-bmc/Config.in"
diff --git a/board/habanalabs/fs1_arc/fs1-local.mk b/board/habanalabs/fs1_arc/fs1-local.mk
index cbd4a6a..1098f5b 100644
--- a/board/habanalabs/fs1_arc/fs1-local.mk
+++ b/board/habanalabs/fs1_arc/fs1-local.mk
@@ -4,4 +4,3 @@
 PID_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/arcpid
 MGMT_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/arcmgmt
 EVENT_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-apps/arcevent
-AGENT_HBM_OVERRIDE_SRCDIR = $(TOPDIR)/../zephyr-plugins
diff --git a/configs/gaudi2_arm_defconfig b/configs/gaudi2_arm_defconfig
index bda35f1..7fcea30 100644
--- a/configs/gaudi2_arm_defconfig
+++ b/configs/gaudi2_arm_defconfig
@@ -66,7 +66,7 @@
 BR2_POST_IMAGE_UBOOT_ITS_TEMPLATE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/uboot_extended.its"
 BR2_PACKAGE_ZEPHYR_AGENT_PCIE=y
 BR2_PACKAGE_ZEPHYR_AGENT_SERDES=y
-BR2_PACKAGE_AGENT_HBM=y
+BR2_PACKAGE_ZEPHYR_AGENT_HBM=y
 BR2_PACKAGE_ZEPHYR_AGENT_FW_LOADER=y
 BR2_PIC_PIE=n
 BR2_SSP_NONE=y
diff --git a/configs/gaudi3_arc_defconfig b/configs/gaudi3_arc_defconfig
index 7744f17..b290190 100644
--- a/configs/gaudi3_arc_defconfig
+++ b/configs/gaudi3_arc_defconfig
@@ -36,6 +36,6 @@
 BR2_RELRO_NONE=y
 BR2_FORTIFY_SOURCE_NONE=y
 BR2_PACKAGE_PLDM_BLOADER=y
-BR2_PACKAGE_AGENT_HBM=y
+BR2_PACKAGE_ZEPHYR_AGENT_HBM=y
 BR2_PACKAGE_ZEPHYR_AGENT_SERDES=y
 BR2_PACKAGE_ZEPHYR_AGENT_PCIE=y
diff --git a/package/agent-hbm/Config.in b/package/agent-hbm/Config.in
deleted file mode 100644
index edc2c63..0000000
--- a/package/agent-hbm/Config.in
+++ /dev/null
@@ -1,5 +0,0 @@
-config BR2_PACKAGE_AGENT_HBM
-        bool "Zephyr agent hbm"
-        help
-          HBM on Zephyr
-
diff --git a/package/agent-hbm/agent-hbm.mk b/package/agent-hbm/agent-hbm.mk
index 7ab572f..9b96eea 100644
--- a/package/agent-hbm/agent-hbm.mk
+++ b/package/agent-hbm/agent-hbm.mk
@@ -1,7 +1,7 @@
 AGENT_HBM_SITE:= agent-hbm
 AGENT_HBM_APP_NAME = hbm
 AGENT_HBM_DEPENDENCIES = host-cmake host-lzop host-lz4
-AGENT_HBM_APP_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init hl_shell mctp pldm
+AGENT_HBM_APP_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init
 
 AGENT_SRC_DIR=$(EMBEDDED_DIR)/zephyr-plugins
 AGENT_MARGIN_TOOL_SRC_DIR=$(AGENT_SRC_DIR)/margin-tools
@@ -23,27 +23,14 @@
 		-I$(AGENT_SRC_DIR)/common/include \
 		-I$(AGENT_SRC_DIR)/common/include/$(HL_DEV)
 
-AGENT_HBM_CONF_APP_CFLAGS +=\
-		-I$(SPECS_EMB_PROJ_DIR) \
-		-I$(SPECS_EMB_PROJ_DIR)/$(DEV_PREFIX) \
-		-I$(SPECS_EMB_PROJ_DIR)/common \
-		-I$(SPECS_EMBEDDED_ROOT) \
-		-I$(DEVICE_SPECS_ROOT) \
-		-I$(DEVICE_SPECS_DIR) \
-		-I$(DEVICE_SPECS_DIR)/asic_reg \
-		-I$(SPECS_ROOT) \
-		-I$(SPECS_ROOT)/$(DEV_PREFIX) \
-		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
-		-I$(SPECS_EXT_ROOT)
-
 AGENT_HBM_CONF_OPTS = $(ZEPHYR_CONF_OPTS)
 
 ifdef BR2_arcle
-AGENT_HBM_CONF_OPTS += -DBOARD=$(ZEPHYR_BOARD_PREFIX)_agent_$(HL_TARGET)
+AGENT_HBM_CONF_OPTS += -DBOARD=hl_$(AGENT_HBM_DEVICE_NAME)_agent_$(HL_TARGET)
 endif
 ifdef BR2_aarch64
 AGENT_HBM_CONF_APP_EXTRA_CFLAGS += -mstrict-align
-AGENT_HBM_CONF_OPTS += -DBOARD=$(ZEPHYR_BOARD_PREFIX)_$(HL_TARGET)
+AGENT_HBM_CONF_OPTS += -DBOARD=hl_$(AGENT_HBM_DEVICE_NAME)_$(HL_TARGET)
 endif
 
 AGENT_HBM_BUILD_DIR = $(@D)
@@ -64,9 +51,15 @@
 AGENT_HBM_CONF_OPTS += -DFAKE_SECURITY=$(FAKE_SECURITY)
 AGENT_HBM_CONF_OPTS += -DBUILD_VERSION="$(AGENT_HBM_APP_VERSION)"
 AGENT_HBM_CONF_OPTS += -DBUILD_MACHINE="$(shell hostname -s)"
-AGENT_HBM_CONF_OPTS += -DCMAKE_C_FLAGS="$(AGENT_HBM_CONF_APP_CFLAGS)"
+AGENT_HBM_CONF_OPTS += -DCMAKE_C_FLAGS="\
+		-I$(SPECS_EMBEDDED_DIR) \
+		-I$(SPECS_EMBEDDED_DIR)/$(DEV_PREFIX) \
+		-I$(SPECS_ROOT) \
+		-I$(SPECS_EXT_ROOT) \
+		-I$(SPECS_ROOT)/$(DEV_PREFIX)/asic_reg \
+		$(AGENT_HBM_CONF_APP_CFLAGS)"
 
-AGENT_HBM_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMB_PROJ_DIR)"
+AGENT_HBM_CONF_OPTS += -DDTS_EXTRA_CPPFLAGS="-I$(SPECS_EMBEDDED_DIR)"
 AGENT_HBM_CONF_OPTS += -DDEVICE=$(AGENT_HBM_DEVICE_NAME)
 AGENT_HBM_CONF_OPTS += -DAPP_NAME=$(AGENT_HBM_APP_PROD_NAME)
 AGENT_HBM_CONF_OPTS += -DAPP_VERSION=$(AGENT_HBM_APP_VERSION)
@@ -120,9 +113,9 @@
 endef
 
 define AGENT_HBM_INSTALL_TARGET_CMDS
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.bin $(ELFS_DIR)/zephyr_agent_$(AGENT_HBM_APP_NAME).elf
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(BINS_DIR)/zephyr_agent_$(AGENT_HBM_APP_NAME).bin
-	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(STATS_DIR)/zephyr_agent_$(AGENT_HBM_APP_NAME).stat
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.bin $(BINARIES_DIR)/zephyr_agent_$(AGENT_HBM_APP_NAME).bin
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.elf $(BINARIES_DIR)/zephyr_agent_$(AGENT_HBM_APP_NAME).elf
+	@$(INSTALL) -D -m 0644 $(@D)/build/zephyr/zephyr.stat $(BINARIES_DIR)/zephyr_agent_$(AGENT_HBM_APP_NAME).stat
 endef
 
 agent-hbm-configure: $(AGENT_HBM_DEPENDENCIES)
diff --git a/package/zephyr/Config.in b/package/zephyr/Config.in
index 66a6d38..5101f8b 100644
--- a/package/zephyr/Config.in
+++ b/package/zephyr/Config.in
@@ -3,6 +3,12 @@
 	help
 	  RTOS for ARC/ARM
 
+config BR2_PACKAGE_ZEPHYR_AGENT_HBM
+        bool "Zephyr agent hbm"
+        select BR2_PACKAGE_ZEPHYR
+        help
+          HBM on Zephyr
+
 config BR2_PACKAGE_ZEPHYR_AGENT_SERDES
         bool "Zephyr agent SerDes"
         select BR2_PACKAGE_ZEPHYR
diff --git a/package/zephyr/zephyr.mk b/package/zephyr/zephyr.mk
index 695bd12..bc1f058 100644
--- a/package/zephyr/zephyr.mk
+++ b/package/zephyr/zephyr.mk
@@ -225,8 +225,10 @@
 $(foreach module,$(ZEPHYR_$(AGENT_UPPER_CASE)_MODULES),$(call DEFINE_MODULE_ENVVARS,$(module),$(ZEPHYR_$(AGENT_UPPER_CASE)_DIR), $(ZEPHYR_$(AGENT_UPPER_CASE)_NAME)))
 endef
 
-ifdef BR2_PACKAGE_AGENT_HBM
-ZEPHYR_COMPONENTS_LIST += agent-hbm
+ifdef BR2_PACKAGE_ZEPHYR_AGENT_HBM
+ZEPHYR_HBM_MODULES = eeprom mbedtls boot_info i2c-indirect soc_init hl_shell mctp pldm
+$(call CREATE_ZEPHYR_MARGIN_TOOL_AGENT,hbm, $(HBM_EXTRA_CONF))
+ZEPHYR_CONFIGURE_CMDS += $(call CREATE_APP_VERSION,hbm,build-agent-hbm)
 endif
 
 ifdef BR2_PACKAGE_ZEPHYR_AGENT_SERDES
